test/struct.jl

```diff
@@ -0,0 +1,40 @@
+using Test, Symbolics
+using Symbolics: StructElement, Struct, operation, arguments
+
+handledtypes = [Int8,
+    Int16,
+    Int32,
+    Int64,
+    UInt8,
+    UInt16,
+    UInt32,
+    UInt64,
+    Float16,
+    Float32,
+    Float64]
+for t in handledtypes
+    @test Symbolics.decodetyp(Symbolics.encodetyp(t)) === t
+end
+
+@variables t x(t)
+a = StructElement(Int8, :a)
+b = StructElement(Int, :b)
+s = Struct([a, b])
+sa = s.a
+sb = s.b
+@test operation(sa) === getfield
+@test arguments(sa) == Any[s, 1]
+@test arguments(sa) isa Any
+@test operation(sb) === getfield
+@test arguments(sb) == Any[s, 2]
+@test arguments(sb) isa Any
+
+sa1 = (setproperty!(s, :a, UInt8(1)))
+@test operation(sa1) === setfield!
+@test arguments(sa1) == Any[s, 1, UInt8(1)]
+@test arguments(sa1) isa Any
+
+sb1 = (setproperty!(s, :b, "hi"))
```

Comment:
shouldn't this error? Or what's the purpose of assigning a string to the Int field here?

Reply:
Type checking is compiler's responsibility. We don't need to check it on the frontend level.